### PR TITLE
fix several streams web-platform tests

### PR DESCRIPTION
**Pull request by anonrig**
Fixes several streams web-platform tests

### CODE REVIEW

**Comment by npaun** [comment]
This fix does exactly what the WPT tests want. Do you think it could break anything else though - I just don't trust floats lol

**Comment by anonrig** [comment]
> This fix does exactly what the WPT tests want. Do you think it could break anything else though - I just don't trust floats lol

It seems this fixes more than I've anticipated. Pushing soon...

**Comment by jasnell** [comment]
Please hold off on merging this. I want to evaluate a couple things here first. Want to see if we can reduce the number of changes in this as I'm not all that happy changing all of the `size_t`s to `double`s.

File: src/workerd/api/streams/common.h [hunk 1]
```diff
@@ -47,9 +47,9 @@ struct ReadResult {
 };
 
 struct StreamQueuingStrategy {
-  using SizeAlgorithm = uint64_t(v8::Local<v8::Value>);
+  using SizeAlgorithm = double(v8::Local<v8::Value>);
 
-  jsg::Optional<uint64_t> highWaterMark;
+  jsg::Optional<double> highWaterMark;
```
Comment (jasnell) [inline src/workerd/api/streams/common.h:52]: This likely needs to be guarded with a compat flag. With this previously mapping to `uint64_t` it means that the implementation would have accepted a `bigint` input which would be rejected after this change. We'll need to make sure we're not breaking anyone with this.

If we want to be reasonably assured that a compat flag is not necessary, then we should first land a PR that adds some logging to see if `bigint` is passed in.

Specifically, with the current code it would be possible to do:

```js
const ws = new WritableStream({}, {
  highWaterMark: 1n,
  size(z) { return 1n; },
});
```

Which I believe would be rejected after this change.

Comment (anonrig) [reply src/workerd/api/streams/common.h:52]: Good catch. I'll just put it behind pedanticWpt compat flag.

Comment (jasnell) [reply src/workerd/api/streams/common.h:52]: We also need to verify that the change in arithmetic won't lead to subtle differences in backpressure signaling here that would break things. We've seen issues previously where observable changes in backpressure signaling have caused workers to break so we just need to be reasonably certain that the changes here aren't going to cause problems. I wouldn't recommend just relying on existing test coverage but instead try to craft some edge case tests to verify. I don't anticipate there *will* be actual issues but we need to be extra cautious in this area.

**Comment by jasnell** [comment]
Now that the statemachine updates have landed, this needs to be revisited.

**Comment by anonrig** [comment]
Will open a separate pull-request.
